### ISSUE TITLE
linter: vitest(valid-expect) doesn't support `expect.assert` methods such as `expect.assert.isDefined`

### ISSUE THREAD

**Issue by goibon** [oxc-project/oxc#27006]
### What version of Oxlint are you using?

1.83.0

### What command did you run?

`oxlint`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```ts
import { defineConfig } from 'oxlint'

export default defineConfig({
  plugins: ['eslint', 'oxc', 'typescript', 'unicorn', 'vitest'],
  rules: { 'vitest/valid-expect': 'error' },
  categories: { correctness: 'off' },
})
```

### What happened?

Using any of the methods on `expect.assert` results in an error like, e.g. calling `expect.assert.isDefined(mightBeUndefined)` results in:

> Expect has an unknown modifier.
help: Is it a spelling mistake? oxc(vitest(valid-expect))

but I would expect it to allow any method under the `expect.assert` property.

Ie. this snippet will trigger it:

```typescript
import { it, expect } from 'vitest'

function foo (shouldReturnString: boolean): string | undefined {
  return shouldReturnString ? '' : undefined
}

it('supports expect.assert', () => {
  const mightBeUndefined = foo(true)

  expect.assert.isDefined(mightBeUndefined) // The offending line
  expect(mightBeDefined.length).toBe(0)
})
```

[Reproducible in this playground](https://playground.oxc.rs/?lintRules=vitest%2Fvalid-expect&options=%7B%22run%22%3A%7B%22lint%22%3Atrue%2C%22formatter%22%3Afalse%2C%22transform%22%3Afalse%2C%22isolatedDeclarations%22%3Afalse%2C%22whitespace%22%3Afalse%2C%22mangle%22%3Afalse%2C%22mangleProps%22%3Afalse%2C%22compress%22%3Afalse%2C%22scope%22%3Atrue%2C%22symbol%22%3Atrue%2C%22cfg%22%3Afalse%7D%2C%22parser%22%3A%7B%22extension%22%3A%22tsx%22%2C%22allowReturnOutsideFunction%22%3Atrue%2C%22preserveParens%22%3Atrue%2C%22allowV8Intrinsics%22%3Atrue%2C%22semanticErrors%22%3Atrue%7D%2C%22linter%22%3A%7B%7D%2C%22formatter%22%3A%7B%22useTabs%22%3Afalse%2C%22tabWidth%22%3A2%2C%22endOfLine%22%3A%22lf%22%2C%22printWidth%22%3A80%2C%22singleQuote%22%3Afalse%2C%22jsxSingleQuote%22%3Afalse%2C%22quoteProps%22%3A%22as-needed%22%2C%22trailingComma%22%3A%22all%22%2C%22semi%22%3Atrue%2C%22arrowParens%22%3A%22always%22%2C%22bracketSpacing%22%3Atrue%2C%22bracketSameLine%22%3Afalse%2C%22objectWrap%22%3A%22preserve%22%2C%22singleAttributePerLine%22%3Afalse%7D%2C%22transformer%22%3A%7B%22target%22%3A%22es2015%22%2C%22useDefineForClassFields%22%3Atrue%2C%22experimentalDecorators%22%3Atrue%2C%22emitDecoratorMetadata%22%3Atrue%2C%22optimizeEnums%22%3Atrue%2C%22optimizeConstEnums%22%3Atrue%7D%2C%22isolatedDeclarations%22%3A%7B%22stripInternal%22%3Afalse%7D%2C%22codegen%22%3A%7B%22normal%22%3Atrue%2C%22jsdoc%22%3Atrue%2C%22annotation%22%3Atrue%2C%22legal%22%3Atrue%7D%2C%22compress%22%3A%7B%7D%2C%22mangle%22%3A%7B%22topLevel%22%3Atrue%2C%22keepNames%22%3Afalse%7D%2C%22mangleProps%22%3A%7B%22include%22%3A%22_%24%22%2C%22reserved%22%3A%5B%5D%2C%22quoted%22%3Afalse%2C%22debug%22%3Afalse%7D%2C%22controlFlow%22%3A%7B%22verbose%22%3Afalse%7D%2C%22inject%22%3A%7B%22inject%22%3A%7B%7D%7D%2C%22define%22%3A%7B%22define%22%3A%7B%7D%7D%7D&code=import+%7B+it%2C+expect+%7D+from+%27vitest%27%0A%0Afunction+foo+%28shouldReturnString%3A+boolean%29%3A+string+%7C+undefined+%7B%0A++return+shouldReturnString+%3F+%27%27+%3A+undefined%0A%7D%0A%0Ait%28%27supports+expect.assert%27%2C+%28%29+%3D%3E+%7B%0A++const+mightBeUndefined+%3D+foo%28true%29%0A%0A++expect.assert.isDefined%28mightBeUndefined%29%0A++expect%28mightBeDefined.length%29.toBe%280%29%0A%7D%29).

The upstream rule looks like it has supported it since [1.6.25](https://github.com/vitest-dev/eslint-plugin-vitest/releases/tag/v1.6.25).
